src/js/placesProvider.js

```diff
@@ -0,0 +1,24 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+/*exported PlacesProvider */
+
+"use strict";
+
+const PlacesProvider = {
+  _placesLinks: [],
+
+  _observers: new Set(),
+
+  setLinks(placesLinks) {
+    this._placesLinks = placesLinks;
+  },
+
+  getLinks() {
+    return Promise.resolve(this._placesLinks);
```

Comment:
It doesn't seem necessary for links to resolve with a promise. It's not an asynchronous operation?